payjoin/src/send/mod.rs

```diff
@@ -358,12 +358,9 @@ pub struct V2PostContext {
 
 #[cfg(feature = "v2")]
 impl V2PostContext {
-    pub fn process_response(
-        self,
-        response: &mut impl std::io::Read,
-    ) -> Result<V2GetContext, ResponseError> {
-        let mut res_buf = Vec::new();
-        response.read_to_end(&mut res_buf).map_err(InternalValidationError::Io)?;
+    pub fn process_response(self, response: &[u8]) -> Result<V2GetContext, ResponseError> {
+        let mut res_buf = [0u8; crate::ohttp::ENCAPSULATED_MESSAGE_BYTES];
+        res_buf[..response.len()].copy_from_slice(response);
         let response = ohttp_decapsulate(self.ohttp_ctx, &res_buf)
```

Comment:
Following the same pattern as V2GetContext::process_response. Also seems better to propagate an error instead of truncating to response.len() in case the length doesn't match the expected size.

```suggestion
        let response_array: &[u8; crate::ohttp::ENCAPSULATED_MESSAGE_BYTES] =
            response
                .try_into()
                .map_err(|_| InternalValidationError::UnexpectedResponseSize(response.len()))?;

        let response = ohttp_decapsulate(ohttp_ctx, response_array)
```